### PR TITLE
py3-bokeh multiversion

### PR DESCRIPTION
**Pull request by dannf**
Add multiversion support to `py3-bokeh`

### CODE REVIEW

**Comment by smoser** [comment]
Just repeating here so you see it.  I have a branch of files converted to mult-version committed that you can start from.
it may or may not help you learn.
https://github.com/chainguard-dev/internal-dev/issues/5334#issuecomment-2460116967

File: py3-bokeh.yaml [hunk 1]
```diff
@@ -38,8 +107,6 @@ pipeline:
   - name: Python Build
     uses: python/build-wheel
 
-  - uses: strip
-
```
Comment (smoser) [inline py3-bokeh.yaml:42]: i think we're leaving these in. they don't do anything if there aren'[t c modules, but its not harmful (and if it was, we need to fix it).

**Review by smoser — CHANGES_REQUESTED** [review]
one change request.

File: py3-setuptools-git-versioning.yaml [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+
+pipeline:
+  - uses: git-checkout
+    with:
+      expected-commit: 1a2b9d0674b881e04b2a1238372b74020558274e
+      repository: https://github.com/dolfinus/setuptools-git-versioning
+      tag: v${{package.version}}
+
+  - name: Python Build
+    uses: python/build-wheel
```
[the remark this answers is not in the document]
Comment (dannf) [reply py3-setuptools-git-versioning.yaml]: nice catch!

File: py3-setuptools-git-versioning.yaml [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+
+pipeline:
+  - uses: git-checkout
+    with:
+      expected-commit: 1a2b9d0674b881e04b2a1238372b74020558274e
+      repository: https://github.com/dolfinus/setuptools-git-versioning
+      tag: v${{package.version}}
+
+  - name: Python Build
+    uses: py/pip-build-install
```
Comment (smoser) [inline py3-setuptools-git-versioning.yaml:105]:
```yaml
        with:
          python: python${{range.key}}
```

Comment (dannf) [reply py3-setuptools-git-versioning.yaml:105]: Actually, do we just need to kill this section now that the builds are happening in the `subpackages:` ?

Comment (smoser) [reply py3-setuptools-git-versioning.yaml:105]: actually, yes. good catch.

Comment (smoser) [reply py3-setuptools-git-versioning.yaml:105]: and can you move the pipeline above the subpackages ? i really wich 'yam' would do all the things, but it does not.

Comment (dannf) [reply py3-setuptools-git-versioning.yaml]: Could we add a jsonschema for the YAML, and and a `pre-commit` hook to validate it? That could enforce ordering, as well as using `yam`

**Review by smoser — APPROVED** [review]
Thanks

**Review by smoser — CHANGES_REQUESTED** [review]
actually. i'm going to ask one more change.
Can you add a variable 'import' and use it?

All of the recently modified py3-* files are using that.  (see [py3-absl-py](https://github.com/wolfi-dev/os/blob/main/py3-absl-py.yaml) for example).  That just gives us better data to work with if we're refactoring things later.

File: py3-setuptools-git-versioning.yaml (hunk 3 above)
Comment (smoser) [reply py3-setuptools-git-versioning.yaml]: huh. i did not realize jsonschema could enforce ordering. I opened https://github.com/chainguard-dev/yam/issues/71 a while ago, I think people would generally be open to computer-enforced formatting (ie, 'go fmt').

Comment (dannf) [reply py3-setuptools-git-versioning.yaml]: You're right - `jsonschema` doesn't enforce ordering (at least, not AFAIK). What I should've said is that I've written `pre-commit` hooks to enforce that YAML keys are sorted in the order that they are defined in a `jsonschema` file. I suppose the schema isn't strictly needed for the sorting, but schema validation is useful as well.

**Review by smoser — APPROVED** [review]

**Pull request merged**